frontend/components/LoginPage.js

```diff
@@ -75,7 +88,12 @@ export default class LoginPage extends React.Component {
 
         // Retrieve the JWT token and decode it
         const decoded = jwtDecode(idToken);
-        const { name } = decoded;
+        const { name, nonce } = decoded;
+        if (nonce != nonce_generated) {
+            Alert.alert("Server Error", "Could not securely login. Please Try again.");
+            this.setState({loading: false});
+            return;
+        } 
 
         // stores the token in SecureStore (only supported on mobile devices)
         if (useProxy) {
```

Comment:
I think `"id_token"`, `"access_token"`, `"expire_time"`, and `"refresh_token"` should be referencing the const in `auth.js`

Reply:
true. I'm actually thinking about refactoring this.